### PR TITLE
feat(spanner): support for enum columns as keys, index and integration tests

### PR DESCRIPTION
**Pull request by harshachinta**
This PR has the following changes,
1. Adds support for Enum columns in primary key and indexes.
2. Integration tests for Parameter Binding.
3. Integration tests for Proto fields and Enum columns as Primary keys.
4. Integration tests for Proto fields and Enum columns as Indexes.

### CODE REVIEW

**Review by rahul2393 — APPROVED** [review]

**Review by gauravpurohit06 — APPROVED** [review]

**Review by gauravpurohit06 — APPROVED** [review]

File: spanner/integration_test.go [hunk 1]
```diff
@@ -2317,6 +2317,165 @@ func TestIntegration_BasicTypes_ProtoColumns_Errors(t *testing.T) {
...
+		},
+		// Filter rows using Proto Message Field Parameterized query
+		{sql: "SELECT SingerId, FirstName, LastName, SingerInfo, SingerGenre FROM Singers WHERE SingerInfo.Nationality=@country",
+			params: map[string]interface{}{
+				"country": "Country3",
+			},
+			want: [][]interface{}{{int64(3), "Singer3", "Singer3"}},
+		},
+	} {
+		ro := client.ReadOnlyTransaction()
```
Comment (olavloite) [inline spanner/integration_test.go:2434]: This (probably) causes a session leak, as this read-only transaction is never closed.

File: spanner/integration_test.go [hunk 2]
```diff
@@ -2317,6 +2317,165 @@ func TestIntegration_BasicTypes_ProtoColumns_Errors(t *testing.T) {
...
+		if err != nil {
+			t.Errorf("%d: Parameterized query returns error %v, want nil", i, err)
+		}
+		if !testEqual(got, test.want) {
+			t.Errorf("%d: got unexpected result from Parameterized query: %v, want %v", i, got, test.want)
+		}
+	}
+
+	// Read all rows based on Proto Enum Primary key column values
+	ro := client.ReadOnlyTransaction()
```
Comment (olavloite) [inline spanner/integration_test.go:2450]: Same here, not closing a read-only transaction will cause a session leak.

**Review by olavloite — APPROVED** [review]

**Review by Shua1 — COMMENTED** [review]
1. Please split the parameter binding part.
2. It seems more accurate, in the descriptor of the PR, to say "support for enum columns..." instead of "support for proto columns".

**Comment by harshachinta** [comment]
> 1. Please split the parameter binding part.
> 2. It seems more accurate, in the descriptor of the PR, to say "support for enum columns..." instead of "support for proto columns".

1. Updated the PR description to make the changes more clearer. As the test for parameter binding is small, raising a separate PR for that would not be useful.
2. Updated the title of the PR.

**Review by Shua1 — COMMENTED** [review]
The PR description says: "support for enum columns as keys, index, parameters".

But we are not adding support for parameters. Right? Please update.

**Comment by harshachinta** [comment]
> The PR description says: "support for enum columns as keys, index, parameters".
> 
> But we are not adding support for parameters. Right? Please update.

Updated the description. Thanks for the review comment, @Shua1

File: spanner/integration_test.go (hunk 1 above)
Comment (Shua1) [reply spanner/integration_test.go]: Are these comments addressed?

Comment (harshachinta) [reply spanner/integration_test.go]: Yes @Shua1. The changes have already been committed.

**Pull request merged**